datus/storage/ext_knowledge/store.py

```
@@ -177,6 +274,20 @@ def delete_knowledge(self, subject_path: List[str], name: str) -> bool:
 
+def gen_subject_item_id(subject_path: List[str], name: str) -> str:
+    """Generate a unique ID from subject_path and name.
+
+    Args:
+        subject_path: Subject hierarchy path (e.g., ['Finance', 'Revenue'])
+        name: Item name
+
+    Returns:
+        ID string in format: "path/component1/component2/.../name"
+    """
+    parts = list(subject_path) + [name]
```

Ruff (0.14.14)

Consider `[*list(subject_path), name]` instead of concatenation

Replace with `[*list(subject_path), name]`

(RUF005)